fe/fe-core/src/main/java/org/apache/doris/nereids/rules/analysis/SubExprAnalyzer.java

```diff
@@ -173,6 +173,9 @@ private void checkRootIsLimit(AnalyzedResult analyzedResult) {
     }
 
     private AnalyzedResult analyzeSubquery(SubqueryExpr expr) {
+        if (cascadesContext == null) {
+            throw new IllegalStateException("Missing CascadesContext");
+        }
```

Comment:
check not null in ctor?

Reply:
In most of UTs, no ConnectContext the test thread. They only want to bind function, and not need to bind Subquery/Slot, and the CascadesContext is not needed.

So we can not check In the constructor.

Reply:
Bind slot/subquery: need context
Bind function: not need context